contrib/epee/include/net/abstract_tcp_server2.inl

```diff
@@ -230,7 +230,8 @@ PRAGMA_WARNING_DISABLE_VS(4355)
     //_dbg3("[sock " << socket_.native_handle() << "] add_ref 2, m_peer_number=" << mI->m_peer_number);
     if(m_was_shutdown)
       return false;
-    m_self_refs.push_back(self);
+    ++m_reference_count;
+    m_self_ref = std::move(self);
```

Comment:
I think m_self_ref will be destroyed by the assignment. Kinda depends if operator== checks for dupes, but I wouldn't want to depend on that.

Reply:
It could be destroyed yes, but at this point you have a guaranteed valid 'self', which holds a reference - so the object couldn't be destroyed by this.

Reply:
That's true, so we can ignore that, OK.